### ISSUE TITLE
Error authenticating with Google: Malformed auth code.

### ISSUE THREAD

**Issue by neok-g** [chrisjshull/homebridge-nest#465]
**Describe the bug**

**To Reproduce**
1. I created a Nest account with my Google account
2. Next I tried to retrieve the refreshtoken via the Using a Google Account - refresh token method
3. Ran the node login.js en navigated to the url and copied the authenticationcode
4. Then I get the error Malformed auth code.

**Expected behavior**
I expected to get the refreshtoken.

Include with your bug report this version info:
```sh
node --version: v14.16.1
homebridge --version: 1.3.6
```

Make sure you have the latest LTS from https://nodejs.org
and the latest packages: `npm upgrade -g homebridge homebridge-nest`

Also include debug log output from startup through seeing the issue:
`DEBUG=* homebridge -D`

**Comment by adriancable** [comment]
@neok-g - this error (which comes from Google) means you are not copying and pasting the auth code correctly. It probably means you accidentally put a space at the beginning or end, didn't get the whole code in, or something like that. If you copy and paste the auth code correctly, everything will work.

**Comment by neok-g** [comment]
@adriancable No because I copied it manually and also via the copy to clipboard button. Both result in the Malformed auth code error. Should I register the Nest account before or after getting the Google refresh token?

**Comment by neok-g** [comment]
Ok when pasting the auth code in Terminal (MacOS) it automatically added a ~ in front and after the auth code. When I removed those the auth code was accepted and I got a refresh token. Restarting the homebridge server did not gave me any (refresh) token errors on startup. So I assume it should be alright now? I expect to receive my Google Nest Protect tomorrow so I should be able to see them in the Apple Home app.
